imagepruner/v1/types.go

```diff
@@ -0,0 +1,99 @@
+// managed by the registry operator.
+type Config struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec ImagePrunerSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status ImagePrunerStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// PrunerHistory is a configuration object for how many
+// successful and failed finished jobs to retain.
+type PrunerHistory struct {
+	// SuccessfulJobsHistoryLimit specifies how many successful finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty" protobuf:"bytes,1,opt,name=successfulJobsHistoryLimit"`
+	// FailedJobsHistoryLimit specifies how many failed finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit,omitempty" protobuf:"bytes,2,opt,name=failedJobsHistoryLimit"`
+}
+
+type ImagePrunerSpec struct {
+	// Schedule specifies when to execute the job
+	// Uses standard cronjob syntax: https://wikipedia.org/wiki/Cron
+	// Defaults to 0 0 * * *
+	// +optional
+	Schedule string `json:"schedule" protobuf:"bytes,1,opt,name=schedule"`
+	// Suspend specifies wether or not to suspend subsequent executions of this cronjob.
+	// Defaults to false
+	// +optional
+	Suspend *bool `json:"suspend,omitempty" protobuf:"bytes,2,opt,name=suspend"`
+	// KeepTagRevisions specifies how many tag revisions to keep.
+	// Defaults to 5
+	// +optional
+	KeepTagRevisions *int `json:"keepTagRevisions,omitempty" protobuf:"bytes,3,opt,name=keepTagRevisions"`
+	// KeepYoungerThan specifies how old an image needs to be for it to be pruned.
+	// Defaults to 96h (96 hours)
+	// +optional
+	KeepYoungerThan time.Duration `json:"keepYoungerThan,omitempty" protobuf:"varint,4,opt,name=keepYoungerThan,casttype=time.Duration"`
+	// Resources defines the resource requests+limits for the image pruner pod.
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,5,opt,name=resources"`
+	// Affinity is a group of node affinity scheduling rules.
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty" protobuf:"bytes,6,opt,name=affinity"`
+	// NodeSelector defines the node selection constraints for the registry
+	// pod.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty" protobuf:"bytes,7,rep,name=nodeSelector"`
+	// Tolerations defines the tolerations for the registry pod.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty" protobuf:"bytes,8,rep,name=tolerations"`
+	// StartingDeadlineSeconds specifies how long of a grace period in seconds
+	// to give the job to start if it misses it's scheduled time.
+	// Defaults to 0
+	// +optional
+	StartingDeadlineSeconds *int64 `json:"startingDeadlineSeconds,omitempty" protobuf:"bytes,9,opt,name=startingDeadlineSeconds"`
```

Comment:
in general i think we're exposing more of the pod template fields than we need to.  did you have specific reasons to expose each of these fields?

Reply:
I was following the enhancement doc? https://github.com/openshift/enhancements/blob/master/enhancements/image-registry/automate-pruning.md